distributed/tests/test_spill.py

```diff
@@ -11,72 +12,132 @@ def test_spillbuffer(tmpdir):
+    assert buf.slow.weight_by_key == {"b": pickle_size}
+    assert buf.slow.total_weight == pickle_size
 
     buf["d"] = d
     assert set(buf.disk) == {"b", "c"}
-    assert buf.spilled_by_key == {"b": s, "c": s}
-    assert buf.spilled_total == s * 2
+    assert buf.slow.weight_by_key == {"b": pickle_size, "c": pickle_size}
+    assert buf.slow.total_weight == pickle_size * 2
 
     # Deleting an in-memory key does not automatically move spilled keys back to memory
     del buf["a"]
     assert set(buf.disk) == {"b", "c"}
-    assert buf.spilled_by_key == {"b": s, "c": s}
-    assert buf.spilled_total == s * 2
+    assert buf.slow.weight_by_key == {"b": pickle_size, "c": pickle_size}
+    assert buf.slow.total_weight == pickle_size * 2
     with pytest.raises(KeyError):
         buf["a"]
 
     # Deleting a spilled key updates the metadata
     del buf["b"]
     assert set(buf.disk) == {"c"}
-    assert buf.spilled_by_key == {"c": s}
-    assert buf.spilled_total == s
+    assert buf.slow.weight_by_key == {"c": pickle_size}
+    assert buf.slow.total_weight == pickle_size
     with pytest.raises(KeyError):
         buf["b"]
 
     # Updating a spilled key moves it to the top of the LRU and to memory
     buf["c"] = c * 2
     assert set(buf.disk) == {"d"}
-    assert buf.spilled_by_key == {"d": s}
-    assert buf.spilled_total == s
+    assert buf.slow.weight_by_key == {"d": pickle_size}
+    assert buf.slow.total_weight == pickle_size
 
     # Single key is larger than target and goes directly into slow
     e = "e" * 500
-    slarge = sizeof(e)
+    pickle_large_size = sum(len(frame) for frame in serialize_bytelist(e))
+
     buf["e"] = e
     assert set(buf.disk) == {"d", "e"}
-    assert buf.spilled_by_key == {"d": s, "e": slarge}
-    assert buf.spilled_total == s + slarge
+    assert buf.slow.weight_by_key == {"d": pickle_size, "e": pickle_large_size}
+    assert buf.slow.total_weight == pickle_size + pickle_large_size
 
     # Updating a spilled key with another larger than target updates slow directly
     buf["d"] = "d" * 500
     assert set(buf.disk) == {"d", "e"}
-    assert buf.spilled_by_key == {"d": slarge, "e": slarge}
-    assert buf.spilled_total == slarge * 2
+    assert buf.slow.weight_by_key == {"d": pickle_large_size, "e": pickle_large_size}
+    assert buf.slow.total_weight == pickle_large_size * 2
+
+
+def test_spillbuffer_maxlim(tmpdir):
+    buf = SpillBuffer(str(tmpdir), target=200, max_spill=600)
+    # Convenience aliases
+    assert buf.memory is buf.fast
+    assert buf.disk is buf.slow
+
+    assert not buf.slow.weight_by_key
+    assert buf.slow.total_weight == 0
+
+    a, b, c, d = "a" * 200, "b" * 100, "c" * 100, "d" * 200
+
+    psize_a = sum(len(frame) for frame in serialize_bytelist(a))  # this is size in disk
+
+    # size of a is bigger than target and is smaller than max_spill, key should be in slow
+    buf["a"] = a
+    assert not buf.memory
+    assert not buf.fast.weights
```

Comment:
please be coherent throughout the test and use either fast/slow or memory/disk everywhere